Introduction to Simulation.ipynb

```diff
@@ -123,34 +310,41 @@
     "import matplotlib.pyplot as plt\n",
     "matplotlib.style.use('default')\n",
     "\n",
+    "#define our model parameters\n",
     "\n",
     "g = -9.8 #gravitation constant\n",
     "dt = 0.0001 #timestep\n",
-    "timesteps  = 100000 #total number of timesteps to consider\n",
-    "\n",
+    "timesteps  = int(8/dt) #total number of timesteps to consider\n",
     "velocity_i = 0.0 #initial velocity\n",
     "height_i = 10.0 #initial height\n",
     "\n",
+    "# define numpy arrays to store the velocity, height and time.\n",
+    "# note, we will define the length of these arrays to match the\n",
+    "# total number of timesteps and initialize all entries to 0\n",
+    "\n",
     "velocity_array = np.zeros(timesteps)\n",
     "height_array  = np.zeros(timesteps)\n",
-    "\n",
     "time_array = np.zeros(timesteps)\n",
     "\n",
+    "# set the height and velocity at time 0 to be the model parameters\n",
+    "# defined above\n",
     "height_array[0] = (height_i)\n",
     "velocity_array[0] = velocity_i\n",
     "\n",
     "current_height = height_i\n",
     "\n",
+    "# integrate the motion of the ball through time\n",
     "for i in range (0, timesteps-1):\n",
     "        \n",
+    "        # kinematic equations\n",
     "        time_array[i+1] = time_array[i] + dt\n",
     "        velocity_array[i+1] = velocity_array[i] + (g*dt)\n",
     "        height_array[i+1] = height_array[i] + 0.5*(velocity_array[i]+velocity_array[i+1])*dt\n",
     "\n",
     "        # are we contacting the ground?\n",
     "        if height_array[i+1] <= 0:\n",
-    "            velocity_array[i+1] =\n",
-    "            height_array[i+1] = height_array[i] + 0.5*(velocity_array[i]+velocity_array[i+1])*dt\n",
+    "            velocity_array[i+1] = \n",
```

Comment:
This should be the solution to Exercise 2. Right now it obviously doesn't run, do you want to leave this down here in exercise 3 filled in correctly so that way if students are having trouble, you can point them here to an already done solution?